test/unit/mdc-menu/simple.foundation.test.js

```diff
@@ -20,23 +20,55 @@ import td from 'testdouble';
 import {captureHandlers, verifyDefaultAdapter} from '../helpers/foundation';
 import {setupFoundationTest} from '../helpers/setup';
 import {createMockRaf} from '../helpers/raf';
-import MDCSimpleMenuFoundation from '../../../packages/mdc-menu/simple/foundation';
-import {cssClasses, strings, numbers} from '../../../packages/mdc-menu/simple/constants';
+import {MDCSimpleMenuFoundation} from '../../../packages/mdc-menu/simple/foundation';
+import {cssClasses, strings, numbers, Corner} from '../../../packages/mdc-menu/simple/constants';
 
 function setupTest() {
   const {foundation, mockAdapter} = setupFoundationTest(MDCSimpleMenuFoundation);
   const size = {width: 500, height: 200};
-  const itemYParams = {top: 100, height: 20};
-  td.when(mockAdapter.hasClass('mdc-simple-menu')).thenReturn(true);
-  td.when(mockAdapter.hasClass('mdc-simple-menu--open')).thenReturn(false);
+  td.when(mockAdapter.hasClass(cssClasses.ROOT)).thenReturn(true);
+  td.when(mockAdapter.hasClass(cssClasses.OPEN)).thenReturn(false);
   td.when(mockAdapter.hasNecessaryDom()).thenReturn(true);
   td.when(mockAdapter.getNumberOfItems()).thenReturn(1);
   td.when(mockAdapter.getInnerDimensions()).thenReturn(size);
-  td.when(mockAdapter.getYParamsForItemAtIndex(0)).thenReturn(itemYParams);
 
   return {foundation, mockAdapter};
 }
 
+// Various anchor dimensions.
+const smallTopLeft = {height: 20, width: 40, top: 20, bottom: 40, left: 20, right: 60};
+const smallBottomLeft = {height: 20, width: 40, top: 920, bottom: 940, left: 20, right: 60};
+const smallBottomRight = {height: 20, width: 40, top: 920, bottom: 940, left: 920, right: 960};
+const smallCenter = {height: 20, width: 40, top: 490, bottom: 510, left: 480, right: 520};
+const smallAboveMiddleLeft = {height: 20, width: 40, top: 400, bottom: 420, left: 20, right: 60};
+const smallBelowMiddleLeft = {height: 20, width: 40, top: 600, bottom: 620, left: 20, right: 60};
+const wideCenter = {height: 20, width: 100, top: 490, bottom: 510, left: 450, right: 550};
+const wideTopLeft = {height: 20, width: 100, top: 20, bottom: 40, left: 20, right: 120};
+
+/**
+ * Initializes viewport, anchor and menu dimensions. Viewport is 1000x1000. Default menu size is 100x200.
+ * @param {Object} mockAdapter Mock double for the adapter.
+ * @param {{height:number, width: number, top: number, bottom: number, left: number, right: number}} anchorDimensions
+ *   Approximate viewport corner where anchor is located.
+ * @param {boolean=} isRtl Indicates whether layout is RTL.
+ * @param {number=} menuHeight Optional height of the menu.
+ */
+function initAnchorLayout(mockAdapter, anchorDimensions, isRtl, menuHeight) {
+  td.when(mockAdapter.hasAnchor()).thenReturn(true);
+  td.when(mockAdapter.getWindowDimensions()).thenReturn({height: 1000, width: 1000});
+  td.when(mockAdapter.getAnchorDimensions()).thenReturn(anchorDimensions);
+
+  if (isRtl) {
```

Comment:
This seems awkward since it's not going to explicitly stub whatsoever if you don't pass `isRtl`, resulting in it being `undefined` (and warranting the coersion here).

It'd make more sense to default `isRtl` to `false` in the function parameters, then simply do `td.when(mockAdapter.isRtl()).thenReturn(isRtl);` (since we always pass a boolean to it).

Reply:
sure, makes sense. I guess I am not used to setting default values on params (yet)!